src/docker/registry.nim

```diff
@@ -39,6 +51,14 @@ const
     "application/vnd.oci.image.config.v1+json": DockerManifestType.config,
   }.toTable()
 
+iterator uses(use: RegistryUse): RegistryUse =
+  ## which uses lookups are applicable for the registry use
+  ## ReadOnly use can only be used for reads
+  ## however ReadWrite use can be used for both
+  yield use
+  if use == RegistryUse.ReadWrite:
```

Comment:
consider renaming to something like "RegistryCapability" or "RegistryPermission". The callsites below are hard to read / reason about

Reply:
its not a registry permission or capability question. some "mirrors" allow you to manually push to them explicitly so the question here is in what context that registry being used, not an absolute capability/permission registry has

for example if you have `1.2.3.4` configured as registry mirror. these are all valid:

```
docker pull alpine # will attempt 1.2.3.4 first
docker pull 1.2.3.4/alpine
docker push alpine # will NOT attempt 1.2.3.4
docker push 1.2.3.4/alpine
```

Reply:
yep. we can rename to `useCase` though. that sounds good and is accurate.

although again, wont do in this pr. will collate all the comments in a follow-up PR